### PR TITLE
Add compute tags LorentzFactor, NonEuclideanMagnitude, and SpecificEnthalpy

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

In addition to the compute tags (which will be useful for follow-ups to #1006):
- Move EoS to Hydro dir
- Fix bug in `CheckWithRandomValues`

Commit order:
1. Fix bug in check with random values
2. Add Lorentz factor function and compute item
3. Move EquationsOfState into PointwiseFunctions/Hydro
4. Move SpecificEnthalpy out of EquationsOfState into Hydro
5. Add specific enthalpy compute tag
6. Add NonEuclideanMagnitude compute tag

### Types of changes:

- [x] Bugfix
- [x] New feature

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by kidder** [comment]
ComputeBla or BlaCompute ?  We have ComputeDiv and ComputeDeriv as compute items that put Compute in the name

**Comment by nilsdeppe** [comment]
I went with `BlaCompute` because then when we list things alphabetically it'll be next to `Bla`. I'm not saying that's correct, but I figured it's reasonable anyway :)

**Comment by nilsdeppe** [comment]
rebased on develop

File: src/DataStructures/Tensor/EagerMath/Magnitude.hpp [hunk 1]
```diff
@@ -71,6 +71,19 @@ struct EuclideanMagnitude : Magnitude<Tag>, db::ComputeTag {
...
+/// The magnitude of a (co)vector with respect to a specific metric
+///
+/// This tag inherits from `Tags::Magnitude<Tag>`
+template <typename Tag, typename MetricTag>
+struct NonEuclideanMagnitude : Magnitude<Tag>, db::ComputeTag {
+  using base = Magnitude<Tag>;
+  static constexpr Scalar<DataVector> (*function)(
+      const db::item_type<Tag>&, const db::item_type<MetricTag>&) = magnitude;
+  using argument_tags = tmpl::list<Tag, MetricTag>;
+};
```
Comment (kidder) [inline src/DataStructures/Tensor/EagerMath/Magnitude.hpp:85]: should there be `using volume_tags = tmpl::list<MetricTag>;`?

Comment (wthrowe) [reply src/DataStructures/Tensor/EagerMath/Magnitude.hpp:85]: I don't think so.  We need the metric on the same mesh as the vector.

**Review by wthrowe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**